eZ/Publish/Core/Persistence/Tests/Legacy/FieldValue/Converter/ImageConverterTest.php

```diff
@@ -0,0 +1,228 @@
+
+    /** @var \eZ\Publish\Core\IO\IOServiceInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $ioService;
+
+    protected function setUp(): void
+    {
+        $this->ioService = $this->createMock(IOServiceInterface::class);
+        $this->urlRedecorator = $this->createMock(UrlRedecoratorInterface::class);
+
+        $this->imageConverter = new ImageConverter(
+            $this->ioService,
+            $this->urlRedecorator
+        );
+    }
+
+    /**
+     * @dataProvider fieldValueToXmlProvider
+     */
+    public function testToStorageValue(FieldValue $fieldValue, string $expectedXml): void
+    {
+        ClockMock::register(ImageConverter::class);
+        ClockMock::withClockMock(true);
+
+        $time = ClockMock::time();
+        $expectedXml = str_replace('{timestampToReplace}', $time, $expectedXml);
+
+        $storageValue = new StorageFieldValue();
+
+        $pathToImg = __DIR__ . '/../_fixtures/ibexa_fav.png';
+        $this
+            ->urlRedecorator
+            ->method('redecorateFromSource')
+            ->willReturn($pathToImg);
+
+        $this->imageConverter->toStorageValue($fieldValue, $storageValue);
+
+        $this->assertEquals(
+            $expectedXml,
+            $storageValue->dataText
+        );
+
+        ClockMock::withClockMock(false);
+    }
+
+    public function fieldValueToXmlProvider(): array
+    {
+        $pathToImg = __DIR__ . '/../_fixtures/ibexa_fav.png';
+        $dir = __DIR__ . '/../_fixtures';
+
+        return [
+            'with_additional_data' => [
+                new FieldValue([
+                    'data' => [
+                        'width' => 100,
+                        'height' => 200,
+                        'alternativeText' => 'test',
+                        'mime' => 'image/png',
+                        'fieldId' => 1,
+                        'uri' => $pathToImg,
+                        'versionNo' => 1,
+                        'languageCode' => 'eng-GB',
+                        'additionalData' => [
+                            'focalPointX' => 50,
+                            'focalPointY' => 100,
+                            'author' => 'John Smith',
+                        ],
+                    ],
+                ]),
+                <<< XML
+<?xml version="1.0" encoding="utf-8"?>
+<ezimage serial_number="1" is_valid="1" filename="ibexa_fav.png"
+    suffix="png" basename="ibexa_fav" dirpath="{$dir}" url="{$pathToImg}"
+    original_filename="ibexa_fav.png" mime_type="image/png" width="100"
+    height="200" alternative_text="test" alias_key="1293033771" timestamp="{timestampToReplace}">
+  <original attribute_id="1" attribute_version="1" attribute_language="eng-GB"/>
+  <information Height="200" Width="100" IsColor="1"/>
+  <additional_data><attribute key="focalPointX">50</attribute><attribute key="focalPointY">100</attribute><attribute key="author">John Smith</attribute></additional_data>
+</ezimage>
+XML,
```

Comment:
https://wiki.php.net/rfc/flexible_heredoc_nowdoc_syntaxes

You can indent this, since we have PHP 7.3+ as dependency.

Reply:
I also thought we can do this, but it seems that indenting XMLs causes issues with linters as XML must start with header and not empty spaces.